common/src/main/java/io/druid/collections/StupidPool.java

```diff
@@ -38,11 +38,23 @@
 
   private final Queue<T> objects = new ConcurrentLinkedQueue<>();
 
+  private final int objectsCacheSize;
+
   public StupidPool(
       Supplier<T> generator
   )
   {
     this.generator = generator;
+    this.objectsCacheSize = Integer.MAX_VALUE;
+  }
+
+  public StupidPool(
+      Supplier<T> generator,
+      int objectsCacheSize
+  )
+  {
+    this.generator = generator;
+    this.objectsCacheSize = objectsCacheSize;
```

Comment:
Suggest naming `objectsCacheCountLimit` or similar. I thought it was bytes to start with.